src/storage/src/source/util.rs

```diff
@@ -144,3 +146,67 @@ where
     // so we are guaranteed that `token` is non-None.
     (data_stream, token.unwrap())
 }
+
+/// lalal
+pub fn async_source<G, D, B, L>(
+    scope: &G,
+    name: String,
+    input: Stream<G, ()>,
+    construct: B,
+) -> (Stream<G, D>, AsyncSourceToken)
+where
+    G: Scope<Timestamp = Timestamp>,
+    D: Data,
+    B: FnOnce(
+        OperatorInfo,
```

Comment:
I see that we use this argument in the implementation to construct a `SyncActivator` and pass it to source readers. Do we need to do that now that it's async? It sounds like all uses of `SyncActivator` would be replaced by waking the waker received from poll context. It will also fix the RA type annotation annoyance you hit :)

Reply:
will add a TODO for this!